recipes/Fine_Tuning_Granite/Finetuning_Granite.ipynb

```diff
@@ -0,0 +1,505 @@
+      ],
+      "metadata": {
+        "id": "wZ1oQO3UkKFm"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "## Evaluation  \n",
+        "  \n",
+        "Finally, we'll evaluate our fine-tuned model by presenting it with the same 'inheritance' prompt we used in the sanity check. This comparison will reveal how the model's understanding has shifted from a programming context to a legal one.  \n",
+        "  \n",
+        "This step demonstrates the power of transfer learning and domain-specific fine-tuning in natural language processing, showing how we can adapt a general-purpose language model to specialized tasks.  "
+      ],
+      "metadata": {
+        "id": "VjhDEDYMHl_S"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "input_text = \"<|user>What does 'inheritance' mean?\\n<|assistant|>\\n\"\n",
+        "inputs = tokenizer(input_text, return_tensors=\"pt\")\n",
+        "\n",
+        "outputs = model.generate(**inputs, max_new_tokens=100)\n",
+        "print(tokenizer.decode(outputs[0], skip_special_tokens=True))"
+      ],
+      "metadata": {
+        "id": "Byruf-ErUx2X"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "### Sample Output\n",
+        "\n",
+        "```\n",
+        "Inheritance refers to the right of a deceased person to inherit property. It is a legal concept that allows the deceased person's estate to be divided among their surviving family members. In the context of real estate, inheritance refers to the right of a deceased person to inherit their property. It is a legal concept that allows the deceased person's estate to be divided among their surviving family members. In the context of real estate, inheritance refers to\n",
+        "\n",
+        "```"
+      ],
+      "metadata": {
+        "id": "aXPG2FQzy9z0"
+      }
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "## Execution Times and Performance Metrics  \n",
+        "  \n",
+        "Throughout this notebook, we've been tracking the time taken for various stages of our process. These execution times provide valuable insights into the computational requirements of fine-tuning a large language model.  \n",
+        "  \n",
+        "We'll summarize the time taken for:  \n",
+        "1. Loading the initial model  \n",
+        "2. Performing the sanity check  \n",
+        "3. Setting up the training environment  \n",
+        "4. The actual training process  \n",
+        "  \n",
+        "Understanding these metrics is crucial for resource planning in machine learning projects. It helps in estimating the time and computational power needed for similar tasks in the future, and can guide decisions about hardware requirements or potential optimizations.  \n",
```

Comment:
Can we say something short about what these particular numbers tell us? Eg., the example metrics taken from running on a T4. Can we extrapolate by model size, data size, and/or GPU size?

Reply:
Candidly, this is just a fluff phrase. I put the timing in because adam was worried about how long it would take.

Reply:
To clarify- I can, or I can drop all the timing stuff. I could go either way with it.

Reply:
I think we could just tone down the language a bit. Otherwise, if we say that "Understanding these metrics is crucial for resource planning", then we should probably explain the metrics, or at least provide a link to where someone could learn to interpret them and do the resource planning.